### PR TITLE
Refactor slider

### PR DESCRIPTION
**Pull request by LidorLapid**

### CODE REVIEW

**Comment by LidorLapid** [comment]
To do:

- [ ] Look into implementing new slider in legacy templates
- [ ] Lazy load slider script using dynamic import
- [ ] Review JS implementation

**Comment by natmunckton** [comment]
Happy with this branch (browser tested and all ok).

File: _includes/elements/picture.html [hunk 1]
```diff
@@ -26,6 +26,6 @@
     {% endif %}
     <!--[if IE 9]></video><![endif]-->
 
-    <img class="picture__image" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" srcset="{{ include.smallSrc }}" alt="{{ include.alt }}">
+    <img class="picture__image" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" srcset="{{ include.smallSrc }}" alt="{{ include.alt }}" {% if include.loading %}loading="{{ include.loading }}"{% endif %}>
```
Comment (wellhairy) [inline _includes/elements/picture.html]: Could this be more succinct by saying ```loading="{{ include.loading | default: "false" }}```

File: _includes/elements/slider.html [hunk 2]
```diff
@@ -1,14 +1,40 @@
-{% assign this = include.params | default: include %}
+<div class="slider {{ include.additionalClasses }}" data-slider>
+  {% for item in include.items %}
+    <div class="slider__item" data-slider-item>
+      {% assign loading = 'eager' %}
 
-<div class="slider{% if this.sliderType %} {{this.sliderType}}{% endif %} {{this.additionalClasses}}">
-  {% for sliderItem in this.sliderItems %}
-    <div class="slider__item" style="background-image: url({{ sliderItem.src }});"></div>
+      {% if forloop.index > 1 %}
```
Comment (wellhairy) [inline _includes/elements/slider.html]: It's the same thing but you could set this up like ```  {% if forloop.first == true %}
    First time through!
  {% else %}
    Not the first time.
  {% endif %}```

File: _includes/components/panel.html [hunk 3]
```diff
@@ -28,6 +28,13 @@
     %}
     {% endif %}
 
+    {% if include.sliderItems %}
+    {% include elements/slider.html
+      items = include.sliderItems
+      additionalClasses = 'panel__slider'
+    %}
+    {% endif %}
```
Comment (wellhairy) [inline _includes/components/panel.html]: This whole chunk feels wrong here. When I look at the panel with this new context I feel it should be separated from this component. what do you think about making panel into an element panel that takes in media and we should create two components, possibly; panel-slider and panel-image.

How are you building this in Squiz? I assume this isn't going to be part of the existing panel element?

**Comment by wellhairy** [comment]
I know we're on a tight time scale but in future can this kind of PR be 2 or 3 different PR's, not 1.

File: _includes/elements/picture.html (hunk 1 above)
Comment (LidorLapid) [reply _includes/elements/picture.html]: Done.

File: _includes/elements/slider.html (hunk 2 above)
Comment (LidorLapid) [reply _includes/elements/slider.html]: Done.

**Pull request merged**